source/_integrations/generic.markdown

```diff
@@ -15,18 +15,13 @@ The `generic` camera platform allows you to integrate any IP camera or other URL
 
 Home Assistant will serve the images via its server, making it possible to view your IP cameras while outside of your network. The endpoint is `/api/camera_proxy/camera.[name]`.
 
-## Configuration
+{% include integrations/config_flow.md %}
 
-To enable this camera in your installation, add the following to your `configuration.yaml` file:
+You must enter a URL in at least one of the fields **Still Image URL** or **Stream Source URL**, the others are optional.
 
-```yaml
-# Example configuration.yaml entry
-camera:
-  - platform: generic
-    still_image_url: http://194.218.96.92/jpg/image.jpg
-```
+[Templates](/topics/templating/) are allowed in the URL fields, which can be used to select different images or parameterize the URL depending on the status of sensors.  Template validity and network access are checked during the configuration steps.
```

Comment:
Also in the UI? That would be confusing when asking for a normal text input...

Reply:
Yes, the fields can still be templates (that is existing functionality). 

If rendering a template fails, an error is displayed. This seems to be a good solution to me... Not sure what a better option would be?

Reply:
I don't think regular text fields are suitable for templates.

Reply:
The template doesn't get rendered in the text field, the user types it there, and sees the result image/stream when the entity is created.  But if it doesn't resolve to a valid image/stream, an error is displayed.

Is there a different type of text field that should be used?  If we import a user's yaml, there could be a template there.

Reply:
I've discussed this a bit with Bram, right now, we indeed only have `object` (which is YAML). We going to look at adding jinja fields support in the near future, so this has to do it for now. 👍